python/ray/data/impl/pipeline_executor.py

```diff
@@ -1,42 +1,36 @@
 from typing import Any, Callable, List, Optional, TYPE_CHECKING
 import time
+import concurrent.futures
 
 import ray
 from ray.data.context import DatasetContext
 from ray.data.dataset import Dataset, T
 from ray.data.impl.progress_bar import ProgressBar, set_progress_bars
-from ray.types import ObjectRef
 
 if TYPE_CHECKING:
     from ray.data.dataset_pipeline import DatasetPipeline
 
 
-# Temporarily use an actor here to avoid ownership issues with tasks:
-# https://github.com/ray-project/ray/issues/20554
-@ray.remote(num_cpus=0, placement_group=None)
-class _StageRunner:
-    def run(self, fn: Callable[[], Dataset[T]], context: DatasetContext) -> Dataset[T]:
-        DatasetContext._set_current(context)
-        try:
-            prev = set_progress_bars(False)
-            # Force eager evaluation of all blocks in the pipeline stage. This
-            # prevents resource deadlocks due to overlapping stage execution
-            # (e.g., task -> actor stage).
-            return fn().force_reads()
-        finally:
-            set_progress_bars(prev)
+def pipeline_stage(fn: Callable[[], Dataset[T]]) -> Dataset[T]:
+    try:
+        prev = set_progress_bars(False)
```

Comment:
We'd probably need to make progress bars a thread-local variable if we want to safely disable it only in this scope